ortools/java/com/google/ortools/Loader.java

```diff
@@ -13,102 +13,75 @@
 
 package com.google.ortools;
 
-import com.sun.jna.Platform;
 import java.io.IOException;
-import java.net.URI;
-import java.net.URISyntaxException;
+import java.io.InputStream;
 import java.net.URL;
-import java.nio.file.FileSystem;
-import java.nio.file.FileSystemAlreadyExistsException;
-import java.nio.file.FileSystems;
-import java.nio.file.FileVisitResult;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.nio.file.SimpleFileVisitor;
-import java.nio.file.attribute.BasicFileAttributes;
-import java.util.Collections;
 import java.util.Objects;
 
-/** Load native libraries needed for using ortools-java.*/
+/**
+ * Helper class to load native libraries needed for using ortools-java.
+ */
 public class Loader {
-  private static final String RESOURCE_PATH = "ortools-" + Platform.RESOURCE_PREFIX + "/";
-
-  /** Try to locate the native libraries directory.*/
-  private static URI getNativeResourceURI() throws IOException {
-    ClassLoader loader = Loader.class.getClassLoader();
-    URL resourceURL = loader.getResource(RESOURCE_PATH);
-    Objects.requireNonNull(resourceURL,
-        String.format("Resource %s was not found in ClassLoader %s", RESOURCE_PATH, loader));
-
-    URI resourceURI;
-    try {
-      resourceURI = resourceURL.toURI();
-    } catch (URISyntaxException e) {
-      throw new IOException(e);
+  private static final String RESOURCE_PREFIX;
+  private static final String RESOURCE_SUFFIX;
+  private static final String[] FILES_TO_EXTRACT;
+  static {
+    RESOURCE_SUFFIX = System.mapLibraryName("test").split("test\\.")[1];
+    switch (RESOURCE_SUFFIX) {
+      case "dll":
+        RESOURCE_PREFIX = "win32-x86-64";
+        FILES_TO_EXTRACT = new String[]{"jniortools"};
+        break;
+      case "so":
+        RESOURCE_PREFIX = "linux-x86-64";
+        FILES_TO_EXTRACT = new String[]{"libjniortools", "libortools"};
+        break;
+      case "dylib":
```

Comment:
this won't work since now we support `darwin-86-64` and `darwin-arm64` (Apple M1) -> you can't rely on the dynamic library extension...